Routers/Graph.cs

```diff
@@ -0,0 +1,186 @@
+﻿/// <summary>
+/// class for building, printing graph and finding the optimal path connecting all vertices of the graph
+/// </summary>
+public class Graph
+{
+    /// <summary>
+    /// a dictionary with a list of neighboring vertices and their distances for a given vertex
+    /// </summary>
+    public Dictionary<int, int> neighbors = new Dictionary<int, int>();
+
+    /// <summary>
+    /// the vertex number of this node
+    /// </summary>
+    public int vertex;
+
+    /// <summary>
+    /// the next node for this vertex is
+    /// </summary>
+    public Graph next;

```

Comment:
Non-nullable field 'next' must contain a non-null value when exiting constructor. Consider declaring the field as nullable.